content/en/blog/_posts/2025-04-23-jobs-successpolicy-goes-ga.md

```diff
@@ -0,0 +1,72 @@
+---
+layout: blog
+title: "Kubernetes 1.33: Job's SuccessPolicy Goes GA"
+date: 2025-04-23
+slug: kubernetes-1-33-jobs-success-policy-goes-ga
+authors: >
+  [Yuki Iwai](https://github.com/tenzen-y) (CyberAgent, Inc)
+---
+
+We're pleased to announce that Job's __Success Policy__ will  graduate to General Availability (GA) as part of Kubernetes v1.33 release.
+
+## About Job's Success Policy
+
+When you run batch workloads with leader-worker patterns, you might want to mark it as succeeded
+even if some the indexes failed failed. Unfortunately, the Kubernetes Job without __Success Policy__ requires in most cases all Pods succeeded for Job succeeded.
+
+Job's __Success Policy__ allows you to specify the early exit criteria for a Job using  the `.spec.successPolicy` 
+field. Which describes a set of rules either using a list of succeeded indexes for a job, or defining a minimal required size of succeeded indexes. 
+
+This feature is especially useful for the scientific simulation workloads and AI/ML or High-Performance computing batch workloads.
+The scientific workloads sometimes perform simulations with different parameters and do not want to care about the number of successful experiments.
+The AI/ML or High-Performance computing workloads sometimes use leader-follower patterns like MPI, in which the leader controls everything, including the followers' lifecycle.
+In this case, the Job failure criteria are leader index failure, and workers failure do not indicate Job failure.
+Additionally, workers do not know when they can terminate themselves.
+
+After Job meets any __Success Policy__, the Job is marked as succeeded, and all Pods are terminated including running ones.
+
+## How it works
+
+The below Job spec with `.successPolicy.rules[0].succeededCount` shows the example of __Success Policy__ feature:
+
+```yaml
+parallelism: 10
+completions: 10
+completionMode: Indexed
+successPolicy:
+  rules:
+  - succeededCount: 1
+```
+
+In this example, the Job is marked as succeeded when one index succeeded regardless of index number.
+Additionally, we can add constraints for index numbers against `succeededCount` to `.successPolicy.rules[0].succeededCount`
+as shown below:
+
+```yaml
+parallelism: 10
+completions: 10
+completionMode: Indexed
+successPolicy:
+  rules:
+  - succeededIndexes: 0,2-4
+    succeededCount: 1
+```
+
+This example shows that the Job is marked as succeeded when one of the indexes 0, 2, 3, or 4 succeeded.
```

Comment:
It might be also a good chance to explain the new `SuccessCriteriaMet` condition.

Reply:
Btw, I have just spot a mistake by an accident. https://github.com/kubernetes/kubernetes/blob/b15dfce6cbd0d5bbbcd6172cf7e2082f4d31055e/pkg/apis/batch/types.go#L263 references `SucceededCriteriaMet`, but it should say `SuccessCriteriaMet` instead.

Reply:
> It might be also a good chance to explain the new `SuccessCriteriaMet` condition.

While re-reading I was thinking about the exactly the same thing. It's definitely a crucial concept to fully understand how a job reaches the early success criteria, even though it's not fully finished.

Reply:
> Btw, I have just spot a mistake by an accident. https://github.com/kubernetes/kubernetes/blob/b15dfce6cbd0d5bbbcd6172cf7e2082f4d31055e/pkg/apis/batch/types.go#L263 references SucceededCriteriaMet, but it should say SuccessCriteriaMet instead.

@atiratree Oh, you're good eye. Thank you for mentioning the typo. I will open a fixing PR.

Reply:
> > It might be also a good chance to explain the new SuccessCriteriaMet condition.
> 
> While re-reading I was thinking about the exactly the same thing. It's definitely a crucial concept to fully understand how a job reaches the early success criteria, even though it's not fully finished.

That sounds reasonable. Let me add description.

Reply:
https://github.com/kubernetes/website/pull/49998#discussion_r2042578882